src/OSPSuite.R/Domain/SimulationBatch.cs

```diff
@@ -128,7 +132,12 @@ public Task<SimulationResults> RunAsync(SimulationBatchRunValues simulationBatch
             _simModelBatch.UpdateParameterValues(simulationBatchRunValues.Values);
             _simModelBatch.UpdateInitialValues(simulationBatchRunValues.MoleculeValues);
             var simulationResults = _simModelBatch.RunSimulation();
-            return _simulationResultsCreator.CreateResultsFrom(simulationResults.Results);
+            var results = _simulationResultsCreator.CreateResultsFrom(simulationResults.Results);
+            if (_simulationBatchOptions.CalculateSensitivity)
```

Comment:
Please rewrite this code. It does not follow our coding standard at all. 
First, return early, then use {} because WHY would you put one if and two foreach without them?

Also why using first. How do you know there is one? 

Last. Why isn't the result prefilled with this stuff already. .TELL Don't ask

Reply:
Actually, this is still a draft... I had to move to something else and did not want to leave things uncommitted. Now I have fixed some things (including the fact that the sensitivities were not properly calculated) and should be better ;)